### ISSUE TITLE
GLEW initalization error ,it comes again

### ISSUE THREAD

**Issue by CharlieLuuke** [deepmind/dm_control#48]
I updated dm_control(with viewer...) today.But when I run sample code in Quickstart, the error comes again.

```python
>>> from dm_control import suite
>>> import numpy as np
>>> 
>>> # Load one task:
... env = suite.load(domain_name="cartpole", task_name="swingup")
>>> 
>>> # Iterate over a task set:
... for domain_name, task_name in suite.BENCHMARKING:
...   env = suite.load(domain_name, task_name)
... 
WARNING: Logging before flag parsing goes to stderr.
F1012 12:47:55.453529 140583399446272 core.py:93] **GLEW initalization error: Missing GL version**
```

However, it works well when I run sample code in "Interactive environment viewer".
I've reinstalled old "dm_control"package，the Quickstart run well(0 error,0 warning). 
I'm so confused why the error comes again.（Write the issue for the first time, please enlighten me.）
(ubantu 16.04)

**Comment by alimuldal** [comment]
Reopening because I'm able to reproduce this error. It only seems to occur when using the GLFW backend (could you try setting the environment variable `MUJOCO_GL=osmesa` before launching the interpreter?). I suspect there may be a race condition involving context cleanup.

**Comment by CharlieLuuke** [comment]
After I added the defensive call, the program works fine. ：）
I think this is just a matter of expediency. I agree that this problem stems from the call to GLFW backend, but this is where I am curious. I don't know how to force the program to use osmesa (failed), I will continue to follow.

**Comment by naeioi** [comment]
we are still experiencing this issue with https://github.com/deepmind/dm_control/commit/0277e4352c57287c9f8497cb600a2bbe6d82ac45 and Mujoco v1.50.

**Comment by alimuldal** [comment]
@naeioi  Could you try adding a defensive call to `env.physics.contexts.mujoco.free()` (as in fe1a361), or else using OSMesa instead of GLFW (`MUJOCO_GL=osmesa`)?

**Comment by naeioi** [comment]
@alimuldal Thanks for advice. I tried calling `env.physics.contexts.mujoco.free()` every time after finishing using `dm_control.suite` but with no luck this exception still occurs nondeterministically (We are running against all available dm_control environments in CI). So we decided to use last stable version https://github.com/deepmind/dm_control/commit/c24ec9f5f3cb3c25c6571c89c9f60bf3350f5711 for now.
